lib/Controller/ConfigController.php

```diff
@@ -231,12 +247,22 @@ public function oauthRedirect(string $code = '', string $state = ''): RedirectRe
 		);
 	}
 
-	private function storeUserInfo(GitlabAccount $account): void {
+	/**
+	 * @param GitlabAccount $account
+	 * @return array{username: string, name: string}
+	 * @throws Exception
+	 */
+	private function getUserInfo(GitlabAccount $account): array {
 		$info = $this->gitlabAPIService->request($account, $account->getUrl(), 'user');
-		if (isset($info['username']) && isset($info['id'])) {
-			$account->setUserInfoName($info['username']);
-			$account->setUserInfoDisplayName($info['name']);
-			$this->accountMapper->update($account);
+		if (isset($info['error'])) {
+			throw new Exception($info['error'], $info['code'] ?? 500);
+		}
+		if (!isset($info['username'])) {
+			throw new Exception('Invalid response from Gitlab API, missing username', 500);
```

Comment:
Please throw some OCS*Exception instead

Reply:
you mean sub-classed exceptions? The controller is not for OCS.

Reply:
Hm ok